core/Tracker/VisitExcluded.php

```diff
@@ -89,8 +89,11 @@ public function isExcluded()
          * @param bool &$excluded Whether the request should be excluded or not. Initialized
          *                        to `false`. Event subscribers should set it to `true` in
          *                        order to exclude the request.
+         * @param Request $request The request object which contains all of the request's information
+         * @param bool|string $userAgent The user-agent used by the visitor
+         * @param bool|string $ip The IP-address of the visitor
          */
-        Piwik::postEvent('Tracker.isExcludedVisit', array(&$excluded));
+        Piwik::postEvent('Tracker.isExcludedVisit', array(&$excluded, $this->request, $this->userAgent, $this->ip));
```

Comment:
$this->request contains only the IP and user agent, when the hook is called during the real visit. When I use the log-import, the values are empty.
This is the reason, why the IP and the user agent are parameters to the constructor:

`public function __construct(Request $request, $ip = false, $userAgent = false)`